docs/Errors.md

```diff
@@ -4,18 +4,49 @@
 
-Fastify follows an all-or-nothing approach and aims to be lean and optimal as much as possible. Thus, the developer is responsible for making sure that the errors are handled properly. Most of the errors are usually a result of unexpected input data, so we recommend specifying a [JSON.schema validation](./Validation-and-Serialization.md) for your input data.
+Given that it is not possible to process all uncaught errors sensibly, the best way to deal with them is to [crash](https://nodejs.org/api/process.html#process_warning_using_uncaughtexception_correctly). 
 
-Fastify tries to catch as many uncaught errors it can without hindering performance. This includes:
+#### Catching Errors In Promises
+In case of promises, make sure to [handle](https://nodejs.org/dist/latest-v8.x/docs/api/deprecations.html#deprecations_dep0018_unhandled_promise_rejections) errors correctly [by making promises safe](https://github.com/mcollina/make-promises-safe).
+
+### Errors In Fastify
+Fastify follows an all-or-nothing approach and aims to be lean and optimal as much as possible. The developer is responsible for making sure that the errors are handled properly. 
+
+#### Errors In Input Data
+Most errors are a result of unexpected input data, so we recommend specifying a [JSON.schema validation](./Validation-and-Serialization.md) for your input data.
+
+#### Catching Uncaught Errors In Fastify
+Fastify tries to catch as many uncaught errors asit can without hindering performance. This includes:
 
 1. synchronous routes, e.g. `app.get('/', () => { throw new Error('kaboom') })`
 2. `async` routes, e.g. `app.get('/', async () => { throw new Error('kaboom') })`
 
-In those two cases, the error will safely be caught by the promise and routed to the default error handler of Fastify for a generic `Internal Server Error` response. For customizing this behaviour, you should use [`setErrorHandler`](./Server.md#seterrorhandler).
+The error in both cases will be caught safely and routed  to Fastify's default error handler for a generic `500 Internal Server Error` response. 
+
+For customizing this behaviour, you should use [`setErrorHandler`](./Server.md#seterrorhandler).
+
+### Errors In Fastify Lifecycle Hooks And A Custom Error Handler
+
+From the [Hooks documentation](https://www.fastify.io/docs/latest/Hooks/#manage-errors-from-a-hook): 
+> If you get an error during the execution of your hook, just pass it to `done()` and Fastify will automatically close the request and send the appropriate error code to the user.
+
+If you have defined a custom error handler for using `setErrorHandler` the error will  be routed there, otherwise it will be routed to Fastify’s generic error handler. 
+
+Some things to consider in your custom error handler: 
+
+- you can `reply.send(data)`  which will be behave as it would be in [regular route handlers](https://www.fastify.io/docs/latest/Reply/#senddata)
```

Comment:
```suggestion
- you can `reply.send(data)` which will be behave as it would be in [regular route handlers](https://www.fastify.io/docs/latest/Reply/#senddata)
```